File: java/poker/src/main/java/com/duchen/poker/model/type/ErrorTypeChecker.java
package com.duchen.poker.model.type;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class ErrorTypeChecker implements TypeChecker {

    @Override
    public CardGroupType getType() {
        return CardGroupType.ERROR;
    }

    @Override
    public List<Integer> getMyPossibleCardCount() {
        return Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20);
    }

    @Override
    public int getMaxCardIfIsMyType(List<Integer> cards) {
        return -1;
    }
}

File: java/poker/src/main/kotlin/poker/model/type/ErrorTypeChecker.kt
package com.duchen.poker.model.type

import java.util.Arrays

class ErrorTypeChecker : TypeChecker {

    override val type: CardGroupType
        get() = CardGroupType.ERROR

    override val myPossibleCardCount: List<Int>
        get() = Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20)

    override fun getMaxCardIfIsMyType(cards: MutableList<Int>): Int {
        return -1
    }
}
